packages/suite-messages/src/messages.ts

```diff
@@ -0,0 +1,32 @@
+import mobileMessagesJson from './mobileMessages.json';
+import webMessagesJson from './webMessages.json';
+import sharedMessagesJson from './sharedMessages.json';
+import { D } from '@mobily/ts-belt';
+
+interface MessageDescriptor<TMessageKey extends string> {
+    id: TMessageKey;
+    defaultMessage: string;
+}
+
+export type MessagesObject<TMessageKey extends string> = Record<
+    TMessageKey,
+    MessageDescriptor<TMessageKey>
+>;
+
+export type SimpleMessagesObject<TMessageKey extends string> = Record<TMessageKey, string>;
+
+const prepareMessages = <TMessageKey extends string>(
+    simpleMessagesObject: SimpleMessagesObject<TMessageKey>,
+): MessagesObject<TMessageKey> =>
+    D.mapWithKey(simpleMessagesObject, (messageKey, messageValue) => ({
+        id: messageKey,
+        defaultMessage: messageValue,
+    }));
+
+export const messages = prepareMessages({ ...sharedMessagesJson, ...webMessagesJson });
+export const mobileMessages = prepareMessages({ ...sharedMessagesJson, ...mobileMessagesJson });
+
+export type MessageId = keyof typeof messages;
+export type MobileMessageId = keyof typeof mobileMessages;
+
+export type MessagesDomain = 'web' | 'mobile';
```

Comment:
I don't like enums, specially in cases when you don't need use it as value. Specifically because it will be translated to some JS nonsense with numbers and functions, even string enums are translated to some strange code. Debuging runtime errors is real pleasure with enums.

I also dislike it for using it as object keys:

```tsx
enum MessageDomainEnum = {
	web: 'web',
	mobile: 'mobile'
}
const myObject: Record<MessageDomainEnum, string> = {
	[MessageDomainEnum.web]: 'hello',
	[MessageDomainEnum.mobile]: 'world'
}

// accessing object later in different file
import { MessageDomainEnum, myObj } from './messages'
console.log(myObject[MessageDomainEnum.web])
```
compared to currently used union:
```tsx
type MessagesDomain = 'web' | 'mobile';
const myObject: Record<MessageDomainUnion, string> = {
	"web": "hello",
	"mobile": "world"
}

// accessing object later in different file
import { myObj } from './myObj'
console.log(myObject.web)
```

I like second version much more, it's cleaner, you don't need extra imports and autocomplete will work exactly same in both cases.

Reply:
Ok!

Reply:
The benefit of enums for me is that you are not operating with a string but with a dedicated data structure, which, true, has to be imported and always used whenever you want to mention it. In code you can clearly see that something is enum, hence has a limited amount of values and has its own meaning. It is also instantly visible if object keys depends on something, since you literally can use the enum at object keys. But it's not required :D Although obviously you wouldn't use it with non-string enums.

![image](https://user-images.githubusercontent.com/45338719/175321393-201ec7bb-08bd-4b46-b382-961e507d7f5d.png) 
![image](https://user-images.githubusercontent.com/45338719/175321569-4b1a5ae4-7691-4a6d-a79b-55daae7e84a1.png)

Reply:
I will just leave last sentence from TS docs about enums here:

> The biggest argument in favour of this format over TypeScript’s enum is that it keeps your codebase aligned with the state of JavaScript, and [when/if](https://github.com/rbuckton/proposal-enum) enums are added to JavaScript then you can move to the additional syntax.

Reply:
I mean, it's true that enums are not in JS but we are using TS anyways, not clear why being "aligned with the state of JavaScript" matters 😅 Just like [mentioned proposal](https://github.com/rbuckton/proposal-enum) says:

> Enums provide a finite domain of constant values that are regularly used to indicate choices, discriminants, and bitwise flags.

Which is what I meant.